src/DwitTech.AccountService.WebApi/Program.cs

```diff
@@ -1,85 +1,96 @@
+using System.Text.Json.Serialization;
+
+namespace DwitTech.AccountService.WebApi
+{
+    public class Program
+    {
+        public static void Main(string[] args)
+        {
+            var builder = WebApplication.CreateBuilder(args);
+
+            // Add services to the container.
+
+            builder.Services.AddControllers()
+                .ConfigureApiBehaviorOptions(options =>
+                {
+                    // To preserve the default behavior, capture the original delegate to call later.
+                    var builtInFactory = options.InvalidModelStateResponseFactory;
+
+                    options.InvalidModelStateResponseFactory = context =>
+                    {
+                        var logger = context.HttpContext.RequestServices
+                                            .GetRequiredService<ILogger<Program>>();
+                        return builtInFactory(context);
+                    };
+                })
+                .AddJsonOptions(options => options.JsonSerializerOptions.DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull);
+
+            // Learn more about configuring Swagger/OpenAPI at https://aka.ms/aspnetcore/swashbuckle
+
+            builder.Services.AddDatabaseService(builder.Configuration);
+
+            builder.Services.AddEndpointsApiExplorer();
+            builder.Services.AddSwaggerGen();
+            builder.Services.AddSwaggerGenNewtonsoftSupport();
+            builder.Services.AddHealthChecks();
+            builder.Services.AddServices(builder.Configuration);
+
+            // Add service and create Policy with options
+            builder.Services.AddCors(options =>
+            {
+                options.AddPolicy("CorsPolicy",
+                    builder => builder.SetIsOriginAllowed(_ => true)
+                        .AllowAnyMethod()
+                        .AllowAnyHeader()
+                        .AllowCredentials());
+            });
+            builder.Services.ConfigureAuthentication(builder.Configuration);
+
+            builder.Services.AddAuthorization();
+            builder.Services.AddMvc();
+            builder.Services.AddControllers();
+            builder.Services.AddHttpContextAccessor();
+            builder.Services.Configure<ForwardedHeadersOptions>(options =>
+            {
+                options.ForwardedHeaders = ForwardedHeaders.XForwardedFor | ForwardedHeaders.XForwardedProto;
+            });
+
+            var app = builder.Build();
+
+            IConfiguration configuration = app.Configuration;
+            IWebHostEnvironment environment = app.Environment;
+
+            // Configure the HTTP request pipeline.
+            app.MapHealthChecks("/health");
+
+            app.UseSwagger();
+            app.UseSwaggerUI();
+
+            app.UseForwardedHeaders();
+
+            app.UseHttpsRedirection();
+
+            app.UseRouting();
+
+            app.UseMiddleware<AuthorizationMiddleware>();
```

Comment:
See? You have this middleware registered here. So no need to invoke it explicitly in the Service layer. Run the app and test your implementation.

Reply:
> See? You have this middleware registered here. So no need to invoke it explicitly in the Service layer. Run the app and test your implementation.

Noted boss